mesonbuild/interpreter/interpreter.py

```diff
@@ -1594,8 +1603,13 @@ def func_build_target(self, node, args, kwargs):
         target_type = kwargs.pop('target_type')
         if target_type == 'executable':
             return self.build_target(node, args, kwargs, build.Executable)
+        elif target_type == 'objects':
+            return self.build_target(node, args, kwargs, build.CompiledObjects)
         elif target_type == 'shared_library':
             return self.build_target(node, args, kwargs, build.SharedLibrary)
+            # TODO: adjust to release
+            FeatureNew('build_target(target_type: \'objects\')',
+                       '0.61.0').use(self.subproject)
```

Comment:
We generally use the `single_use` factory method here. I think you can also pass the `node` argument to `FeatureNew` to get line speicifc error information.

Reply:
Could use bit of advice on how to implement compiled_objects install:

Here is my attempt:
https://github.com/piotrrak/meson/tree/compiled_objects_install

Got bit stuck here.

Cleanest way I see is to be able to create list of objects from sources in `CompiledObjects` and specify it as `target.outputs`. That would mean moving logic of object list generation from `NinjaBackend::generate_target` to `BuildTarget`. This feels ackward and ninja backend specific;

It is more involved than I would expected due to:
- vala
- cython
- generated sources

Second option is to lift it Backend and special case for CompiledObjects in Backend::generate_target install.

Anyone has insights about how this should be solved?